src/array.cpp

```diff
@@ -48,6 +49,13 @@ namespace sparrow
         return array_element(*p_array, index);
     }
 
+    void array::slice(size_type start, size_type end)
```

Comment:
I would prefer slice to return a array which is a view to the original array.
ie auto sliced_arr = arr.slice(start, end);
With the view function of the arrow_proxy it should be possible to create a cheap "view-copy" of the array and than change the offset and size there

Reply:
Now we have arr.slice and arr.slice_view